libbeat/processors/add_process_metadata/gosysinfo_provider.go

```diff
@@ -39,6 +40,18 @@ func (p gosysinfoProvider) GetProcessMetadata(pid int) (result *processMetadata,
 		return nil, err
 	}
 
+	var userInfo types.UserInfo
+	userInfo, err = proc.User()
+	if err != nil {
+		return nil, err
+	}
+
+	var u *user.User
+	u, err = user.LookupId(userInfo.UID)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is adding two new failure conditions to the `add_process_metadata` processor. If there's an error accessing the process' owner, or looking up the user ID, the whole process enrichment operation will fail.

This is dangerous as this code is expected to run in all supported OS and under unprivileged conditions. It's possible for example for this code to only be successful when run with high privileges.

I think it's better for GetProcessMetadata to still succeed if any of this calls fails and silently leave the Username field empty.

Reply:
done